src/fragments/lib/graphqlapi/js/working-with-files.mdx

```diff
@@ -0,0 +1,910 @@
+The Storage and GraphQL API categories can be used together to associate a file, such as an image or video, with a particular record. For example, you might create a `User` model with a profile picture, or a `Post` model with an associated image. With Amplify's GraphQL API and Storage categories, you can reference the file within the model itself to create an association.
+
+To get started, run the following in an existing Amplify project:
+
+```bash
+npm i @aws-amplify/ui-react  #For Authenticator component
+amplify add auth             #Select default configuration
+amplify add storage          #Select S3 with read/write access
+amplify add api              #Select Cognito User Pool for authorization type
+```
+
+When prompted, use the following schema:
+
+```graphql
+type Song @model @auth(rules: [{ allow: public}]) {
+  id: ID!
+  name: String!
+  covertArtKey: String #Set as optional to allow adding file after initial create
+}
+```
+
+Save the schema and run `amplify push` to deploy the changes.
+
+To use files and attachments you need authorization credentials for reading and writing to Amazon S3, which `amplify add auth` configures in the default setting along with a Cognito User Pool.
+
+<Callout>
+
+The Storage and API categories govern data access based on their own authorization patterns. Adding an `@auth` directive to the API schema does not guard against access in the Storage category. When you run `amplify add storage`, the CLI will configure appropriate IAM policies on the bucket using a Cognito Identity Pool role. You will then have the option of adding CRUD (Create, Update, Read and Delete) based permissions as well, so that Authenticated and Guest users will be granted limited permissions within these levels. For more on configuring Storage authorization levels, see the [Storage documentation](https://docs.amplify.aws/lib/storage/configureaccess/q/platform/js/).
+
+</Callout>
+
+For more on configuring API authorization (separate from Storage authorization), see the [API documentation](https://docs.amplify.aws/lib/graphqlapi/authz/q/platform/js/).
+
+<Callout>
+
+For the complete working example, including required imports, obtaining the file from the user, and React component state management, see the [Complete Example](#complete-example) below.
+
+</Callout>
+
+## Create a record with an associated file
+
+First upload the file to Storage, then run the following mutation with the GraphQL API to create a record and associate the file with the record.
+
+<Callout>
+
+Note: In the case of adding a file to an existing record (such as when a user uploads a profile picture to an existing profile), first create the record and exclude the optional file reference, then follow the steps under [Add or update a file for an associated record](#add-or-update-a-file-for-an-associated-record) below.
```

Comment:
@cwomack - there are two ways to create a record with an associated file: 1) upload the Storage file, then create the API record with the file association, and 2) create the API record, upload the Storage file, then update the API record to create the association. I put the callout before the snippet because I would like to make it clear to the customer that there are two workflows, and I'd like to contain that guidance in one place. If we put this below the snippet, the customer may not clearly understand that there are two initial workflows.